runtime/src/snapshot_bank_utils.rs

```diff
@@ -875,6 +876,50 @@ mod tests {
         Arc::into_inner(bank).unwrap()
     }
 
+    /// Convenience function to create an fastboot snapshot out of any Bank, regardless of
+    /// state.  The Bank will be frozen during the process.
+    ///
+    /// Requires:
+    ///     - `bank` is complete
+    fn bank_to_fastboot_snapshot(
+        bank_snapshots_dir: impl AsRef<Path>,
+        bank: &Bank,
+        snapshot_version: SnapshotVersion,
+    ) -> agave_snapshots::Result<()> {
+        assert!(bank.is_complete());
```

Comment:
Thinking on this more, let's move it to another PR. This change (and the tests below) isn't required to add SnapshotKind::Fastboot.